File: app/src/main/java/com/wavesplatform/wallet/v1/ui/dex/details/chart/CandleTouchListener.java
package com.wavesplatform.wallet.v1.ui.dex.details.chart;

import android.graphics.Matrix;
import android.view.MotionEvent;
import android.view.View;

import com.github.mikephil.charting.charts.BarLineChartBase;
import com.github.mikephil.charting.data.BarLineScatterCandleBubbleData;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.interfaces.datasets.IBarLineScatterCandleBubbleDataSet;
import com.github.mikephil.charting.listener.BarLineChartTouchListener;

public class CandleTouchListener extends BarLineChartTouchListener {
    /**
     * Constructor with initialization parameters.
     *
     * @param chart               instance of the chart
     * @param touchMatrix         the touch-matrix of the chart
     * @param dragTriggerDistance the minimum movement distance that will be interpreted as a "drag" gesture in dp (3dp equals
     */
    public CandleTouchListener(BarLineChartBase<? extends BarLineScatterCandleBubbleData<? extends IBarLineScatterCandleBubbleDataSet<? extends Entry>>> chart, Matrix touchMatrix, float dragTriggerDistance) {
        super(chart, touchMatrix, dragTriggerDistance);
    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        boolean res =  super.onTouch(v, event);
        switch (event.getAction() & MotionEvent.ACTION_MASK) {
            case MotionEvent.ACTION_UP:
                ((OnCandleGestureListener)mChart.getOnChartGestureListener()).onChartUp(event);
        }
        return res;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/ui/custom/CandleTouchListener.kt
package com.wavesplatform.wallet.v2.ui.custom

import android.graphics.Matrix
import android.view.MotionEvent
import android.view.View

import com.github.mikephil.charting.charts.BarLineChartBase
import com.github.mikephil.charting.data.BarLineScatterCandleBubbleData
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.interfaces.datasets.IBarLineScatterCandleBubbleDataSet
import com.github.mikephil.charting.listener.BarLineChartTouchListener

class CandleTouchListener
/**
 * Constructor with initialization parameters.
 *
 * @param chart instance of the chart
 * @param touchMatrix the touch-matrix of the chart
 * @param dragTriggerDistance the minimum movement distance that will be interpreted as a "drag" gesture in dp (3dp equals
 */
(chart: BarLineChartBase<out BarLineScatterCandleBubbleData<out IBarLineScatterCandleBubbleDataSet<out Entry>>>, touchMatrix: Matrix, dragTriggerDistance: Float) : BarLineChartTouchListener(chart, touchMatrix, dragTriggerDistance) {

    override fun onTouch(v: View?, event: MotionEvent): Boolean {
        val res = super.onTouch(v, event)
        when (event.action and MotionEvent.ACTION_MASK) {
            MotionEvent.ACTION_UP -> (mChart.onChartGestureListener as OnCandleGestureListener).onChartUp(event)
        }
        return res
    }
}
